src/link/patch.c

```diff
@@ -340,6 +340,46 @@ static int32_t computeRPNExpr(struct Patch const *patch,
 #undef popRPN
 }
 
+void patch_CheckAssertions(struct Assertion *assertion)
+{
+	verbosePrint("Checking assertions...");
+	initRPNStack();
+
+	uint8_t failures = 0;
+
+	while (assertion) {
+		if (!computeRPNExpr(&assertion->patch, assertion->section)) {
+#define ERR_MSG(extra) (assertion->message[0] ? "%s: " extra "assertion failure: %s" \
+					      : "%s: " extra "assertion failure"), \
+		       assertion->patch.fileName, assertion->message
+
+			switch ((enum AssertionType)assertion->patch.type) {
+			case ASSERT_FATAL:
+				errx(1, ERR_MSG("fatal "));
+			case ASSERT_ERROR:
+				fprintf(stderr, ERR_MSG(""));
+				failures++;
+				break;
+			case ASSERT_WARN:
+				warnx(ERR_MSG("ignored "));
+				break;
+			}
+#undef ERR_MSG
+		}
+		struct Assertion *next = assertion->next;
+
+		free(assertion);
+		assertion = next;
+		if (failures == 100)
```

Comment:
The threshold is far too low, and it should be configurable anyway. (Even valgrind's infamous 10 million limit can be configured via a command-line flag!) For example, an automated build would probably want _all_ assertions, even if there's a quattordecillion of them; you'd want to see as many as you can in the recorded build output.

I'd recommend a default of 999 and a command-line flag to set the limit to anything else, including 0 to disable it entirely.

(Also, it's generally a better idea to do `>=` checks for this kind of thing, just in case there's a bug somewhere.)